### PR TITLE
#129 filter search products

### PR DESCRIPTION
**Pull request by quocminhh47**
#129 

- filter products by brand name: 
https://user-images.githubusercontent.com/103627485/193188096-35eb5cd6-dbaa-4379-98c4-2624de99b12d.mp4

find product by filter and searching (combination)

https://user-images.githubusercontent.com/103627485/193188591-b61befe8-c7f1-4896-83a5-c854a1c65eed.mp4

### CODE REVIEW

File: product/src/main/java/com/yas/product/service/ProductService.java [hunk 1]
```diff
@@ -46,15 +46,12 @@ public ProductService(ProductRepository productRepository, MediaService mediaSer
...
     }
 
-    public List<ProductListVm> getProducts() {
-        return productRepository.findAll().stream()
-                .map(ProductListVm::fromModel)
-                .toList();
-    }
-
-    public ProductListGetVm getAllProducts(int pageNo, int pageSize) {
+    public ProductListGetVm getProductsByBrandOrName(int pageNo, int pageSize, String productName, String brandName) {
```
Comment (thiennn) [inline product/src/main/java/com/yas/product/service/ProductService.java:49]: getProductsWithFilter might be a better name, similar for repository

File: product/src/main/java/com/yas/product/service/ProductService.java [hunk 2]
```diff
@@ -46,15 +46,12 @@ public ProductService(ProductRepository productRepository, MediaService mediaSer
...
         Pageable pageable = PageRequest.of(pageNo, pageSize);
-        Page<Product> productPage = productRepository.findByOrderByIdAsc(pageable);
+        Page<Product> productPage = productRepository.findProductsByFilterOrSearching(
+                productName.trim().toLowerCase(),
```
Comment (thiennn) [inline product/src/main/java/com/yas/product/service/ProductService.java:52]: I think we don't need to call toLowerCase here?

**Review by thiennn — CHANGES_REQUESTED** [review]

File: product/src/main/java/com/yas/product/service/ProductService.java (hunk 2 above)
Comment (quocminhh47) [reply product/src/main/java/com/yas/product/service/ProductService.java:52]: i want to search product by name which means ignore case. For instance: we can find "Samsung" by enter "samsung" keyword

File: product/src/main/java/com/yas/product/service/ProductService.java (hunk 1 above)
Comment (quocminhh47) [reply product/src/main/java/com/yas/product/service/ProductService.java]: i changed the function name. Thanks for your suggestion !

File: product/src/main/java/com/yas/product/service/ProductService.java (hunk 2 above)
Comment (thiennn) [reply product/src/main/java/com/yas/product/service/ProductService.java:52]: This should be handled by collation in database. I though it was case insensitive already

Comment (quocminhh47) [reply product/src/main/java/com/yas/product/service/ProductService.java]: > This should be handled by collation in database. I though it was case insensitive already

i got you. Thanks for your suggestion, i fixed the code . Please have a check

**Review by thiennn — APPROVED** [review]

**Pull request merged**